packages/iov-ethereum/src/utils.ts

```diff
@@ -22,3 +23,37 @@ export function decodeHexQuantityNonce(hexString: string): Nonce {
   const nonce = decodeHexQuantity(hexString);
   return new Int53(nonce) as Nonce;
 }
+
+export function encodeQuantity(value: number): string {
+  if (Number.isNaN(value)) {
+    throw new Error("Input is not a number");
+  }
+  try {
+    return "0x" + new BN(value).toString(16);
+  } catch {
+    throw new Error("Input is not a safe integer");
+  }
+}
+
+export function encodeQuantityString(value: string): string {
+  if (!value.match(/^[0-9]+$/)) {
+    throw new Error("Input is not a valid string number");
+  }
+  return "0x" + new BN(value).toString(16);
+}
+
+export function stringDataToHex(stringData: string): string {
+  let hex = "";
+  for (let i = 0; i < stringData.length; i++) {
+    const n = stringData.charCodeAt(i).toString(16);
+    hex += n.length < 2 ? "0" + n : n;
+  }
+  return "0x" + hex;
```

Comment:
This is a reimplementation of `Encoding.toHex(Encoding.toUtf8(stringData))`. Please use those well-tested methods